crates/wallet/src/lib.rs

```diff
@@ -75,5 +92,262 @@ pub trait AlphaNetWalletApi {
+            // whitelisted address
+            (false, Some(TxKind::Call(addr))) => {
+                let state =
+                    self.provider.latest().map_err(|_| AlphaNetWalletError::InternalError)?;
+                let delegated_address = state
+                    .account_code(addr)
+                    .ok()
+                    .flatten()
+                    .and_then(|code| match code.0 {
+                        Bytecode::Eip7702(code) => Some(code.address()),
+                        _ => None,
+                    })
+                    .unwrap_or_default();
+
+                // not a whitelisted address, or not an eip-7702 bytecode
+                if delegated_address == Address::ZERO
+                    || !valid_delegations.contains(&delegated_address)
+                {
+                    return Err(AlphaNetWalletError::IllegalDestination.into());
+                }
+            }
+            // if it's an eip-7702 tx, let it through
+            (true, _) => (),
+            // create tx's disallowed
+            _ => return Err(AlphaNetWalletError::IllegalDestination.into()),
+        }
+
+        // set nonce
+        let tx_count = EthState::transaction_count(
+            &self.eth_api,
+            NetworkWallet::<Ethereum>::default_signer_address(&self.wallet),
+            Some(BlockId::pending()),
+        )
+        .await
+        .map_err(Into::into)?;
+        request.nonce = Some(tx_count.to());
+
+        // set chain id
+        request.chain_id = Some(self.chain_id);
+
+        // set gas limit
+        let estimate =
+            EthCall::estimate_gas_at(&self.eth_api, request.clone(), BlockId::latest(), None)
+                .await
+                .map_err(Into::into)?;
+        request = request.gas_limit(estimate.to());
+
+        // set gas fees
+        let (max_fee_per_gas, max_priority_fee_per_gas) =
+            LoadFee::eip1559_fees(&self.eth_api, None, None)
+                .await
+                .map_err(|_| AlphaNetWalletError::InvalidTransactionRequest)?;
+        request.max_fee_per_gas = Some(max_fee_per_gas.to());
+        request.max_priority_fee_per_gas = Some(max_priority_fee_per_gas.to());
+
+        // build and sign
+        let envelope =
+            <TransactionRequest as TransactionBuilder<Ethereum>>::build::<EthereumWallet>(
+                request,
+                &self.wallet,
+            )
+            .await
+            .map_err(|_| AlphaNetWalletError::InvalidTransactionRequest)?;
+
+        // this uses the internal `OpEthApi` to either forward the tx to the sequencer, or add it to
+        // the txpool
+        //
+        // see: https://github.com/paradigmxyz/reth/blob/b67f004fbe8e1b7c05f84f314c4c9f2ed9be1891/crates/optimism/rpc/src/eth/transaction.rs#L35-L57
+        EthTransactions::send_raw_transaction(&self.eth_api, envelope.encoded_2718().into())
+            .await
+            .map_err(Into::into)
```

Comment:
we should log the error here, because this can fail if we're out of funds for example or out of txpool slots